### DISCUSSION TITLE
MultiSelect functionality / general select direction

### DISCUSSION THREAD

**Discussion by blite** [filamentphp/filament#3661] [General]
Taken from the demo I find this select functionality very confusing to use. 

https://demo.filamentphp.com/shop/products/1/edit?activeRelationManager=0

https://user-images.githubusercontent.com/442226/186506606-93b3f2e1-891f-4caa-b729-0b4302a4eb79.mov

1. If a user doesn't know what category they want, it seems there is no way to know what the other categories are.  If you just click the field it doesn't load other options until you start typing.  Then when you backspace out it keeps the options you searched for on the last character, leading you to believe maybe you are loading options for all

2. I cannot get multiselect on a relationship working with preloaded options.  If I follow the directions in https://github.com/filamentphp/filament/discussions/2347#discussioncomment-2722551 it doesn't load existing relationship as values.  Could be I'm missing something and should dive into the code here more, but this might should be documented if other fixes aren't appropriate.  

3. For some time I have noticed UI bugs with the space between the selection options and the field. I find a connected layout like below easier to use:
<img width="538" alt="Screen Shot 2022-08-24 at 12 29 03 PM" src="https://user-images.githubusercontent.com/442226/186506667-00ff29e8-4d7f-4d1e-aa64-97bb9d03991e.png">

4. Option pagination really needs to be supported and probably loops back to no 1 in that if it were supported you could make the default to load options before searching.  Select2 has had this option for years and I've seen it in use many times https://select2.org/data-sources/ajax#request-parameters

5. Likely needs a limit option which is supported by choices.js. Example limit tags to 10.  This particular complaint might could be solved with a recipe to do this, and no specific options.

6. Labels for groups of options https://select2.org/appearance#labels

I would be willing to undertake some of these fixes, but would need to know which way filament wants to go on these issues.

**Comment by blite** [comment]
adding
```
->preload(true)
```
fixes my issue in no 2 but it appears this isn't documented at all. If this is not an internal api I would suggest the demo should be using it as well for categories.

**Reply by jo-argon7** [reply]
in filament/tables **getTableFilters** function, can you use preload() as well? 
this does not seem to work: 
`SelectFilter::make('myfilter')->multiple()->relationship('relationshipname','label')->preload()`
:-(

**Comment by danharrin** [comment]
Thanks for the feedback, I would be open for a design proposal for 3.

4 - I don't know if Choices supports this, and the implementation may be way too complex for us

5 - We do have limit option in place, have you tried it? Unclear if you mean options limit or selected options limit

6 - Not sure what the usage syntax for this would be, we currently just use key => value arrays of options
